service/models.py

```diff
@@ -96,17 +102,20 @@ def deserialize(self, data):
             self.product_id=data["product_id"]
             self.condition = self.Condition(data["condition"])
 
-            if "name" in data and data["name"] is not None:
-                self.name = data["name"]
-
-            if "quantity" in data and data["quantity"] is not None:
-                self.quantity = data["quantity"]
-
-            if "reorder_quantity" in data and data["reorder_quantity"] is not None:
-                self.reorder_quantity = data["reorder_quantity"]
-
-            if "restock_level" in data and data["restock_level"] is not None:
-                self.restock_level = data["restock_level"]
+            if data.get("name"):
+                if isinstance(data.get("name"), str):
+                    self.name = data.get("name")
+                else:
+                    raise TypeError
+
+            for field in ["quantity", "reorder_quantity", "restock_level"]:
+                if data.get(field):
+                    if not isinstance(data.get(field), int):
+                        raise TypeError
+                    elif data.get(field) < 0:
```

Comment:
why are we raising a valueerror? if the data.get(field) is empty shouldn't we stick to what is already present in the database?

Reply:
If the field is empty, `data.get(field)` returns a `None`. In which case we are raising a `TypeError` -> `DataValidationError` which is already present.

When the type is as expected (int), however when the value is negative, it may not be appropriate to raise the same `TypeError`. So raised a `ValueError` (already present) which in turn raise `OutOfRangeError` (analogous to `DataValidationError`)

Reply:
gotcha!